nncf/quantization/algorithms/bias_correction/algorithm.py

```diff
@@ -229,60 +229,62 @@ def _get_subgraph_data_for_node(self, node: NNCFNode, nncf_graph: NNCFGraph) ->
-            node = edge.from_node
-            input_id = (edge.to_node.node_name, edge.input_port_id)
-
-            # Since we need to find the inputs for the subgraph,
-            # we can take only those layers for which we have already collected statistics.
-            if input_id in self._collected_stat_inputs_map:
-                activation_id = self._collected_stat_inputs_map[input_id]
-                if activation_id not in subgraph_output_ids:
-                    subgraph_input_ids.append(input_id)
-                    return
+        def fill_subgraph_output_ids(main_node: NNCFNode):
+            edges_queue = deque(nncf_graph.get_output_edges(main_node))
+            while edges_queue:
+                edge = edges_queue.pop()
+                node = edge.to_node
+                if node in visited_nodes:
+                    continue
+                visited_nodes.append(node)
+
+                input_id = (node.node_name, edge.input_port_id)
+                output_id = (edge.from_node.node_name, edge.output_port_id)
+
+                # If we found a node with bias, we have to collect it as a statistic node,
+                # and its input for _collected_stat_inputs_map,
+                # which will be used during the collection of statistics for the next node.
+                if self._is_node_correctable(node, nncf_graph):
+                    subgraph_output_ids.append(output_id)
+                    self._collected_stat_inputs_map[input_id] = output_id
+
+                    statistic_nodes.append(edge.from_node)
+                    continue
+
+                edges_queue.extend(nncf_graph.get_output_edges(node))
+
+        def fill_subgraph_input_ids(main_node: NNCFNode):
+            edges_queue = deque(nncf_graph.get_input_edges(main_node))
+            while edges_queue:
+                edge = edges_queue.pop()
+                node = edge.from_node
+                input_id = (edge.to_node.node_name, edge.input_port_id)
 
-            if node in visited_nodes:
-                return
-            visited_nodes.append(node)
+                # Since we need to find the inputs for the subgraph,
+                # we can take only those layers for which we have already collected statistics.
+                if input_id in self._collected_stat_inputs_map:
+                    activation_id = self._collected_stat_inputs_map[input_id]
+                    if activation_id not in subgraph_output_ids:
+                        subgraph_input_ids.append(input_id)
+                        continue
 
-            for input_edge in nncf_graph.get_input_edges(node):
-                fill_subgraph_input_ids(input_edge)
+                if node in visited_nodes:
+                    continue
+                visited_nodes.append(node)
+                edges_queue.extend(nncf_graph.get_input_edges(node))
 
+        nncf_logger.info(f"Processing subgraph data for {node.node_name} node")
```

Comment:
Could you share the output of the BC before and after this change, to look a difference?

Reply:
That was added for debugging purposes. Removed.